packages/components/src/presentation-components/properties/inputs/AsyncSelect.tsx

```diff
@@ -114,17 +115,44 @@ function ClearIndicator<TOption, IsMulti extends boolean = boolean>({ children:
 
 /** @internal */
 export function AsyncSelect<OptionType, Group extends GroupBase<OptionType>, Additional>(props: AsyncPaginateProps<OptionType, Group, Additional, true>) {
+  const [dropdownUp, setDropdownUp] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+  const menuObserver = useRef<IntersectionObserver>();
+  const selectRef = useRef<SelectInstance<OptionType, true, Group>>(null);
   const divRef = useRef<HTMLDivElement>(null);
   const { ref: resizeRef, width } = useResizeObserver();
+
+  const onMenuOpen = () => {
+    const observeOnscreen: IntersectionObserverCallback = (entries: IntersectionObserverEntry[]) => {
```

Comment:
I think we can take a simpler approach here. I think menu max-height is set to 300px so it should be enough to check if there are enough space below select for max heigh menu.

Reply:
Switched to only checking for max-height.